include/storage/dedicated_io.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef OSRM_STORAGE_DEDICATED_IO_HPP_
+#define OSRM_STORAGE_DEDICATED_IO_HPP_
+
+#include "contractor/query_edge.hpp"
+#include "extractor/extractor.hpp"
+#include "extractor/original_edge_data.hpp"
+#include "extractor/query_node.hpp"
+#include "util/exception.hpp"
+#include "util/fingerprint.hpp"
+#include "util/simple_logger.hpp"
+#include "util/static_graph.hpp"
+#include "storage/io.hpp"
+
+#include <boost/filesystem/fstream.hpp>
+#include <boost/iostreams/seek.hpp>
+
+#include <cerrno>
+#include <cstring>
+#include <tuple>
+#include <type_traits>
+
+namespace osrm
+{
+namespace storage
+{
+namespace dedicated_io
+{
+
+
+// To make function calls consistent, this function returns the fixed number of properties
+inline std::size_t readPropertiesCount() { return 1; }
+
+struct HSGRHeader
+{
+    std::uint32_t checksum;
+    std::uint64_t number_of_nodes;
+    std::uint64_t number_of_edges;
+};
+
+// Reads the checksum, number of nodes and number of edges written in the header file of a `.hsgr`
+// file and returns them in a HSGRHeader struct
+inline HSGRHeader readHSGRHeader(io::FileReader &input_file)
+{
+    const util::FingerPrint fingerprint_valid = util::FingerPrint::GetValid();
+    const auto fingerprint_loaded = input_file.ReadOne<util::FingerPrint>();
+    if (!fingerprint_loaded.TestGraphUtil(fingerprint_valid))
+    {
+        util::SimpleLogger().Write(logWARNING) << ".hsgr was prepared with different build.\n"
+                                                  "Reprocess to get rid of this warning.";
+    }
```

Comment:
The problem is that the FileReader doesn't do this specific fingerprint test: `if (!fingerprint_loaded.TestGraphUtil(fingerprint_valid))` so we have to do it here.